tests/containers/docker_image.pm

```diff
@@ -46,7 +48,6 @@ sub run {
             exec_on_container($iname, $runtime, 'cat /etc/os-release');
         }
     }
-    scc_restore_docker_image_credentials() if (get_var('SCC_DOCKER_IMAGE') && !$test_data->{host_os});
```

Comment:
Please keep this here, just remove the `&& !$test_data->{host_os}` part.

Reply:
This line has been moved inside the `if (check_host_os())` L45

Reply:
It should be on the same level as `scc_apply_docker_image_credentials()` is.

Reply:
oh you mean outside of the forloop, right? sure i will fix.

Reply:
Yes :) Thank you

Reply:
done

Reply:
i am reopening this as i just noticed that the `(get_var('SCC_DOCKER_IMAGE')` is used inside the function so it does not make sense here. And i think i need to adjust the `scc_apply_docker_image_credentials` as well

Reply:
Why not? If the `SCC_DOCKER_IMAGE` is set, then `scc_apply_docker_image_credentials` and `scc_restore_docker_image_credentials ` are executed. The value of `SCC_DOCKER_IMAGE` is also used in `scc_apply_docker_image_credentials` as it is supposed to be.

Reply:
my concern is that we have `scc_restore_docker_image_credentials() if (get_var('SCC_DOCKER_IMAGE'));` which inside the  `scc_restore_docker_image_credentials` the same exact variable is checked again. but run only if `is_sle`

Reply:
I see! I think it's not needed.
```perl
sub scc_restore_docker_image_credentials {
    assert_script_run "cp /etc/zypp/credentials.d/SCCcredentials{.bak,}" if (is_sle() && get_var('SCC_DOCKER_IMAGE'));
}
```
should be enough.